wallet-api/test/Generators.hs

```diff
@@ -0,0 +1,108 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS -fplugin=Language.Plutus.CoreToPLC.Plugin -fplugin-opt Language.Plutus.CoreToPLC.Plugin:dont-typecheck #-}
+module Generators(
+    -- * Mockchain
+    Mockchain(..),
+    genMockchain,
+    -- * Transactions
+    genInitialTransaction,
+    genValidTransaction,
+    -- * Useful scripts
+    unitData,
+    unitRedeemer,
+    emptyValidator
+    ) where
+
+import           Crypto.Hash           (Digest, SHA256, hash)
+import           Data.Map              (Map)
+import qualified Data.Map              as Map
+import           Data.Maybe (fromMaybe)
+import           Hedgehog
+import qualified Hedgehog.Gen          as Gen
+import qualified Hedgehog.Range        as Range
+
+import           Language.Plutus.TH    (plutusT)
+
+import           Wallet.Emulator.Types
+import           Wallet.UTXO
+
+-- | Blockchain for testing the emulator implementation and traces.
+--   
+--   To avoid having to rely on functions from the implementation of
+--   wallet-api (in particular, `Wallet.UTXO.unspentOutputs`) we note the
+--   unspent outputs of the chain when it is first created.
```

Comment:
I think we may need to rely on that eventually.

Reply:
Ok, but we still need to test `unspentOutputs`, and in general we can save some time by not having to scan the entire chain each time we need an unspent txn output

Reply:
Sure. But we're going to need to do something like this if we want to generate chains with more than one txn, otherwise we'll end up implementing `updateUtxo` ,which is probably equally complex.